server/src/main/java/com/exacaster/lighter/rest/PermanentSessionController.java

```diff
@@ -0,0 +1,43 @@
+package com.exacaster.lighter.rest;
+
+import com.exacaster.lighter.application.Application;
+import com.exacaster.lighter.application.SubmitParams;
+import com.exacaster.lighter.application.sessions.SessionService;
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.annotation.Body;
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.http.annotation.Delete;
+import io.micronaut.http.annotation.PathVariable;
+import io.micronaut.http.annotation.Put;
+import io.micronaut.http.annotation.Status;
+import io.micronaut.validation.Validated;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import io.swagger.v3.oas.annotations.tags.Tags;
+
+import javax.validation.Valid;
+
+@Validated
+@Tags(@Tag(name = "Sessions"))
+@Controller("/lighter/api/permanentSessions")
```

Comment:
I'm not sure if we should have a dedicated controller & ApplicationType for Permanent sessions.

Maybe it would make sense to have a single type "SESSION", as we do now, and just extend "SubmitParams" body for session creation with some optional "isPermanent" field? In that case, a single API controller would be enough.

To manage Permanent sessions, we could introduce a separate database table, something like "application_features", which would have reference to the application, as well some "management type = PERMANENT" field. That table could be reused for other similar features in the future, for example retry policies, repeat policies for the batch jobs, etc..

/cc @Minutis

Reply:
> I'm not sure if we should have a dedicated controller & ApplicationType for Permanent sessions.
> 
> Maybe it would make sense to have a single type "SESSION", as we do now, and just extend "SubmitParams" body for session creation with some optional "isPermanent" field? In that case, a single API controller would be enough.
> 
Can also be done and then have an extra "if" to decide what to do. Please note that creating perm session is pretty simple and does not differ a lot from creating a regular session. Deleting it could require some extra logic (soft delete) and it would require extra "if". Why not make it in a controller ? 

> To manage Permanent sessions, we could introduce a separate database table, something like "application_features", which would have reference to the application, as well some "management type = PERMANENT" field. That table could be reused for other similar features in the future, for example retry policies, repeat policies for the batch jobs, etc..
YAGNI
> /cc @Minutis

Reply:
I do not think we need an extra “if” for deletion, the logics behind deletion should be the same for all applications. If we go with soft delete - we should add it to all types of applications.

if we go with a separate controllers for sessions & permanent sessions, we’d have to duplicate statement API under a different path, otherwise it would look quite strange, if you create a session through “/permannentSessions”, but post  statements to it through “/sessions/:id/statements”. Also some changes would have to be imlemented on the Lighter UI.

Reply:
> I do not think we need an extra “if” for deletion, the logics behind deletion should be the same for all applications. If we go with soft delete - we should add it to all types of applications.

Ok. i assumed wrongly soft-delete would be only for permanent session - to find out if what we have in YAML has not been deleted before via REST API.

> if we go with a separate controllers for sessions & permanent sessions, we’d have to duplicate statement API under a different path, otherwise it would look quite strange, if you create a session through “/permannentSessions”, but post statements to it through “/sessions/:id/statements”. Also some changes would have to be imlemented on the Lighter UI.

true.

Reply:
I added soft-delete and removed a dedicated controller for perm sessions